### PR TITLE
A few additions

### PR DESCRIPTION
**Pull request by LaurentTreguier**
I think using a `dev` branch is better than directly pushing to `master`, so I'll propose changes from here before merging. This PR brings a few goodies:
- Full support for autocompleting dub.sdl based projects using the new `dub convert` command
- Finally got around adding linting with dscanner
- Added messages to the status bar when fetching/building packages: at first launch the extension may have to fetch and build all used packages, and as it can take a while it's easy to think the extension isn't working
- The `dub` command now has a configurable path
- Fixes here and there

@mattiascibien is it ok to merge ?

### CODE REVIEW

File: src/dcd/server.ts [hunk 1]
```diff
@@ -46,7 +48,7 @@ export default class Server extends vsc.Disposable {
             additionsImports.push('-I' + item);
         })
 
-        if (process.platform === 'windows') {
+        if (process.platform === 'win32') {
             additionsImports.push('-IC:\\D\\dmd2\\src\\phobos');
```
Comment (mattiascibien) [inline src/dcd/server.ts]: Hi, isn't possibile here to detect this in some way?

Comment (LaurentTreguier) [inline src/dcd/server.ts]: @mattiascibien I add somewhat forgotten about this, maybe adding this to the extension configuration would be better

Comment (mattiascibien) [inline src/dcd/server.ts]: @LaurentTreguier nice. Thank you :+1:

Comment (LaurentTreguier) [inline src/dcd/server.ts]: @mattiascibien I've added some configuration. On Linux and OS X the paths can vary... Thankfully, there is always a configuration file (generally `/etc/dmd.conf`) which contains the paths. On Windows however there is no configuration file, so import paths are directly configured. They are always the same though, from what I've seen.
So Windows and POSIX systems aren't exactly configured the same way, but it should work out of the box on any system.

**Comment by mattiascibien** [comment]
@LaurentTreguier seems fine to me... I guess it also closes #5. I will merge this and then we can publish 0.6.0 as it is preetty useable now (for dub commands we can delay it to 0.7.0)

**Comment by mattiascibien** [comment]
@LaurentTreguier for a future we can also try to use https://github.com/dlang/dmd/blob/master/ini/windows/bin/sc.ini#L7 to detect the imports (like dmd.conf on Linux).

**Comment by LaurentTreguier** [comment]
@mattiascibien :+1: `sc.ini` seems to be in the same directory as `dmd.exe`, so locating it shouldn't be hard

**Comment by mattiascibien** [comment]
@LaurentTreguier I have moved the repo to an ORg.

**Comment by mattiascibien** [comment]
@LaurentTreguier I will run a few test and then try to release the new extension on the Marketplace if anything is right.

**Comment by mattiascibien** [comment]
@LaurentTreguier seems fine to me... Maybe we need some more extensive testing on Windows.

**Comment by mattiascibien** [comment]
@LaurentTreguier the only thing i see is that I get a warning in the debug console about a `INVALID result from completion provider. expected CompletionItem-array or CompletionList but got: Object` when having an empty list of completion items.

EDIT: I also tend to get a lot of spawned dcd-server.exe for I do not know why. My dub project has only one file. http://screencast.com/t/iExap0RwEq

**Comment by LaurentTreguier** [comment]
@mattiascibien The warning should be gone now. As for the dcd-server.exe spam... It's weird. It should detect if another one is already running... How do you quit the extension host when testing ? It  looks like closing the window won't dispose of all the `context.subscriptions`, while stopping the debug from the original window will kill all processes associated with the debugged extension. (and the dcd-server.exe with it)

**Comment by mattiascibien** [comment]
@LaurentTreguier I did not quit at all... They get spawned during one session only.

**Comment by LaurentTreguier** [comment]
@mattiascibien I trust you have the latest version of VSCode installed ? The same thing happened here: https://github.com/jcward/vscode-haxe/issues/3 and it was because of VSCode itself.
https://github.com/Microsoft/vscode/issues/1298 is also still open...

**Comment by mattiascibien** [comment]
@LaurentTreguier I guess so.. Will check tomorrow... I will also try the insider build in case.

**Comment by mattiascibien** [comment]
@LaurentTreguier i have tested on insiders build and seems better and I get only two dcd-server processes. I am not sure about the behavior anyway.

EDIT: nope. spamming ctrl+space spawns a lot of dcd-server processes.

**Comment by LaurentTreguier** [comment]
@mattiascibien I've just looked again at the code, and the only way to spawn multiple dcd-server.exe's is if the client can't contact the server. In this case the extension assumes the server is not started and tries to start it again... Can your dcd-client.exe contact the dcd-server.exe ?

**Comment by mattiascibien** [comment]
@LaurentTreguier in most cases I think so... But maybe this is due to spamming multiple CTRL+Space when the completion windows is not fully populated yet. Could this be the issue?

**Comment by LaurentTreguier** [comment]
@mattiascibien It looks like when the server is launched but is not ready to provide completion CTRL+space makes the extension try to restart the server. Normally the new server should exit with an error because another one is already launching... But it doesn't. Maybe checking if a server is already launched should be in the code to correct this

**Comment by LaurentTreguier** [comment]
@mattiascibien It should be fixed now... At least it seems to be working fine on windows for me

**Comment by mattiascibien** [comment]
@LaurentTreguier seems fine also for me... I will do a merge now.

**Comment by mattiascibien** [comment]
@LaurentTreguier online now https://marketplace.visualstudio.com/items?itemName=dlang-vscode.dlang

**Comment by LaurentTreguier** [comment]
@mattiascibien It looks like after moving the repo, the publisher is now dlang-vscode, and it's now considered a seperate extension... And thus the older one doesn't get updated

**Comment by mattiascibien** [comment]
@LaurentTreguier damn... well, i guess it is better this way as it is not under my publisher name (as I did not much recently)

**Comment by LaurentTreguier** [comment]
@mattiascibien is there a way to update the old one just to say it's deprecated with a popup or something ?

[EDIT] or even unpublishing it, it looks like it's possible to unpublish a VSCode extension

**Comment by mattiascibien** [comment]
I think not... The old one cannot be found anymore on the repo.

**Comment by LaurentTreguier** [comment]
Isn't the extension publisher the one logged in with `vsce login` ? If so it would still be possible

**Comment by mattiascibien** [comment]
I changed it and deleted the old one actually. I guess it is the reason for not having the old one present. Did not know that the ID changed

**Comment by LaurentTreguier** [comment]
I just tried going to `.vscode/extenstions/mattiasscibien.dlang-0.5.0` and executed `vsce unpublish`. It asked for an access token. So using your access token, you could still unpublish the dlang extension installed from VSCode

**Comment by mattiascibien** [comment]
write me an email at mattias@mattiascibien.net so I can give you the token (you will also be able to manage the new one)

**Pull request merged**